### ISSUE TITLE
Only able to get 100 reviews

### ISSUE THREAD

**Issue by alfredtangsw** [woctezuma/download-steam-reviews#9]
Hello @woctezuma, I am trying to download all reviews for multiple apps on steam, and I wanted to understand how this tool works. 

I use the sample provided in your readme:
[appID = 573170] expected #reviews = 284

I am running this in JupyterLab.

The problem I have now is that the JSON file returned only 100 rows. How do I get it to retrieve all reviews?

Also, please let me know if the answer was already in [here](https://github.com/woctezuma/download-steam-reviews/issues/7). Thanks!

**Comment by woctezuma** [comment]
The tool is using [Steam API](https://partner.steamgames.com/doc/store/getreviews) to collect reviews. There are API constraints, e.g. each API request can at most retrieve 100 reviews. So, the tool has to generate multiple requests and aggregate the result. If you only have 100 reviews, without any error message, it is because the tool in charge of the aggregation believes that it is collecting redundant reviews.

1. One explanation would be that you have set

```python
request_params['filter'] = 'all'
```

Make sure it is either set to `'recent'` or `'updated'`.

2. Another explanation would be that there already exists a file with some recent reviews on your disk. This would happen if you have downloaded reviews in the past, and re-run the program now, so it is only fetching the reviews posted since then. It allows to download new reviews without starting from scratch, but this assumes that the previous download process was error-free and completed its task.

If there is an issue with the JSON file, make sure to delete it prior to running the program again.

**Comment by alfredtangsw** [comment]
Thanks for getting back to me so quickly. Yes, I got the 100 reviews with no error message.

I checked the JSON file and it did indeed also only have 100 reviews stored in it, so it may be the request_params. I will try that and see how that goes.

I am aware that it can only get 100 reviews per request, so I was wondering if I needed to tell it to run multiple requests. I will get back to you after I try the 'recent' or 'updated' params.

Thanks!

**Comment by woctezuma** [comment]
I have just run this code in a new local environment:

```bash
pip install steamreviews
```

```python
import steamreviews

app_id = 573170
review_dict, query_count = steamreviews.download_reviews_for_app_id(app_id)
```

And I do get 284 reviews in the JSON file.

![all the reviews](https://i.imgur.com/imGpWhE.png)

Maybe try with another appID.

**Comment by alfredtangsw** [comment]
You're right, maybe it was just a quirk of that particular game. It works fine for the app_id you just mentioned :)

**Comment by alfredtangsw** [comment]
Hey @woctezuma would you mind helping me test something out? I am trying to get all English reviews for appid 730 (CS GO) for the past 365 days (my project requires really big data). I did indeed manage to get reviews up to that date, but there were only ~97,000 English ones out of an expected 600,000+ ones.

The params used were:
request_params['filter'] = 'recent'
request_params['language'] = 'english'
request_params['day_range'] = '365'
request_params['purchase_type'] = 'steam'

I am running another set of queries now to see if I get different results. The params are:

request_params['filter'] = 'updated'
request_params['language'] = 'english'
request_params['day_range'] = '365'

The report I am seeing is: [appID = 730] expected #reviews = 1005360

So far, most attempts I have done end with me getting only a small percentage of the expected review counts, maybe 20%. I'm wondering if this is because of reviews deleted by Steam, e.g. spam reviews. 

Could advise me please?
-----
Off-topic, but I have a suggestion/request: When the query encounters a 502 error, can we get the script to sleep for a few seconds before trying again? I really need a big dataset for my project (1 year's worth of reviews) and the probability of a 502 error happening before completion is quite high when going back 365 days...

**Comment by woctezuma** [comment]
> So far, most attempts I have done end with me getting only a small percentage of the expected review counts, maybe 20%. I'm wondering if this is because of reviews deleted by Steam, e.g. spam reviews.

I believe the difference is explained by:
- deleted reviews (in case they are offensive),
- hidden reviews (you can share reviews with friends, rather than making them publicly visible).

That being said, 20% seems very low.

**Comment by woctezuma** [comment]
> When the query encounters a 502 error, can we get the script to sleep for a few seconds before trying again? I really need a big dataset for my project (1 year's worth of reviews) and the probability of a 502 error happening before completion is quite high when going back 365 days...

Status code 502 is for "Bad Gateway". It is mostly due to the Steam servers being saturated during summer sales. If you try again outside of the sale window, it should be fine.

**Comment by woctezuma** [comment]
An attempt to tackle 502 status code is available in v0.8.2.

Feel free to open a new issue if needs be. It is easier to track than adding comments to closed issues.
